safehttp/plugins/hsts/hsts.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package hsts
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/google/go-safeweb/safehttp"
+)
+
+// Plugin implements automatic HSTS functionality.
+type Plugin struct {
+	maxAge            uint64
+	includeSubDomains bool
+	preload           bool
+}
```

Comment:
setters and getters do nothing on this type, so does the constructor, I'd suggest creating somth with a valid zero-value. E.g:

```suggestion
type Plugin struct {
	maxAge            uint64
	disableIncludeSubDomains bool
	preload           bool
}
```
And when the plugin runs if maxage is 0 you use a safe default.

Then you can remove constructor, getters and setters and put the opportune comments on the fields of this struct.

Reply:
Nice, I will do this!

But we might want `maxAge=0` as a legitimate value since it is used to expire and disable HSTS:
```
Should it be necessary to disable Strict Transport Security, setting the max-age to 0 (over a https connection) will immediately expire the Strict-Transport-Security header, allowing access via http.
```
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security

So we can't replace a 0 with a default safe value. How could we deal with this without having a constructor which sets `maxAge`?